### ISSUE TITLE
Reproducing Fig. 2b (EMPIAR-10076) from CryoDRGN-AI — “thin-shell” reconstruction issue

### ISSUE THREAD

**Issue by Ao-King** [ml-struct-bio/drgnai#27]
I am attempting to reproduce the results in Fig. 2b (EMPIAR-10076) from your paper, “CryoDRGN-AI: neural ab initio reconstruction of challenging cryo-EM and cryo-ET datasets” (Nature Methods, 2025). Following the official instructions at https://ez-lab.gitbook.io/cryodrgn-ai, we were able to largely reproduce the clustering and latent-space structure shown in Fig. 2b.

However, we consistently encounter a “thin-shell” structure during 3D reconstruction. Specifically, the HPS phase already yields a hollow shell, and subsequent SGD refinement only sharpens this shell rather than producing a volumetric density.

For your reference, I have attached:

    Our reproduction configuration file (drgnai-configs.yaml)

    Training logs (including the HPS→SGD transition and loss curves)

    Latent-space clustering results (UMAP/PCA, comparable to Fig. 2b)

    Example density maps illustrating the “thin-shell” outcome

I would be grateful for your advice on two points:

    1. HPS quality and initialization. Could the quality of pose estimation in the HPS phase be the main reason for the “thin-shell” collapse during reconstruction? If so, how did your team obtain a good initial result at this stage? Are there key parameters—such as rotational grid resolution, frequency band limits, or the switching schedule to SGD—that you recommend tuning?

    2. batch_size_latent_optimization. I noticed your publicly available output config includes batch_size_latent_optimization: 256, which is not mentioned explicitly in the paper. Does this parameter play a critical role in successful reconstruction, and could it be contributing to our difficulty reproducing the 3D results?

Note. Our hardware setup differs slightly from that described in the paper. We do not have A100 GPUs; instead, we are using eight RTX 4090 cards (48 GB each) to match the batch size reported. Could this hardware difference significantly affect reconstruction quality?

[training(1).log](https://github.com/user-attachments/files/23113319/training.1.log)
[training.log](https://github.com/user-attachments/files/23113320/training.log)
[drgnai-configs(1).yaml](https://github.com/user-attachments/files/23113321/drgnai-configs.1.yaml)
[drgnai-configs.yaml](https://github.com/user-attachments/files/23113322/drgnai-configs.yaml)

<img width="601" height="695" alt="Image" src="https://github.com/user-attachments/assets/0a331ed3-9e2e-4467-b4b7-f4a37f9a3640" />
<img width="605" height="698" alt="Image" src="https://github.com/user-attachments/assets/74a29c8e-42be-404b-a180-952dbe9973ef" />
<img width="1024" height="790" alt="Image" src="https://github.com/user-attachments/assets/5769b918-e678-43ee-9796-3fc5a57baccf" />
<img width="516" height="389" alt="Image" src="https://github.com/user-attachments/assets/80429ac4-2d13-4a23-af60-d7daed365df7" />
<img width="616" height="695" alt="Image" src="https://github.com/user-attachments/assets/27a610ad-214c-42ee-9616-41df667dba3e" />
<img width="605" height="695" alt="Image" src="https://github.com/user-attachments/assets/7d6ecffb-a0a3-4578-9d33-76205661f98d" />

**Comment by michal-g** [comment]
Hi, the key thing here is that the data is being inverted (`invert_data: True`), assuming originally inverted images; however the 50S dataset being used here is one of the only ones with originally *not* inverted images and thus `invert_data` needs to be set to `False`. Although you are seeing a hollow shell, I can recognize that the model does appear to be doing a decent job of learning the volume based on the features appearing on the inside of this shell (comparing to instances when I myself have forgotten to set `invert_data: False` when running on 50S).

Nevertheless, especially when running on novel datasets other than 50S, we recommend experimenting with different values of hyper-parameters, especially batch sizes and learning rates, to see if they yield better results, as the default values are not necessarily tuned to be optimal for all cases.

By the way, you can invert these outputs yourself to produce the correct volumes using tools such as `cryodrgn_utils invert_contrast` and the `--invert` flag to the `cryodrgn analyze` command if you don't feel like rerunning the model!

Let us know if you have any other questions!

**Comment by Ao-King** [comment]
Thanks for the clarification — that really helped me resolve the issue!

I just wanted to mention that the line in the documentation:
invert_data: [bool, default = True] Flag for not inverting input data (e.g. for EMPIAR-10076)
might be a bit ambiguous. It could be clearer if it explicitly stated what happens when invert_data is set to True or False.

Thanks again for your help!
